### DISCUSSION TITLE
WebGL Support for Big Data

### DISCUSSION THREAD

**Discussion by sonsuzdejavu** [d3/d3#3492] [Q&A] [answered]
Can we use webgl with D3.js?

Look at this performance test, https://ahoak.github.io/renderer-benchmark/

And this demo https://sanddance.js.org/app/

**Comment by curran** [comment, marked as answer]
You could use D3 to do the data transformations, and use WebGL to do the rendering, for sure.

**Reply by sonsuzdejavu** [reply]
Thank you, do you know any example?

**Reply by Fil** [reply]
https://peterbeshai.com/blog/2017-05-26-beautifully-animate-points-with-webgl-and-regl/

**Reply by sonsuzdejavu** [reply]
Cool article, thanks.

BTW, examples of D3's are much better on bl.ocks.org then Observablehq.

https://sanddance.js.org/app/
BUT, still 3d interaction part is missing... :/

**Reply by Fil** [reply]
see also https://blog.scottlogic.com/2020/05/01/rendering-one-million-points-with-d3.html

**Reply by dolev146** [reply]
have you successfully implemented them in your code?

**Comment by khuongtp** [comment]
A little bit off topic but isn't the canvas supposed to be faster than svg?, cuz on my computer, the benchmark says otherwise

**Reply by curran** [reply]
Canvas is generally faster than SVG for rending many primitives.

What benchmark are you using?

**Reply by khuongtp** [reply]
I used the link above
![image](https://github.com/d3/d3/assets/62872729/c3622f97-b622-4971-9ecf-ea6bd5a3af45)
![image](https://github.com/d3/d3/assets/62872729/c4d5adb9-12e6-4417-a722-5310eb3ad28b)

**Reply by mcFrax** [reply]
This doesn't seem to be a fair comparison between svg and canvas. The animation is a simple transition, which in svg is likely implemented with native transition (i.e. no JS runs to render the frames), while for canvas each frame needs processing in JS - though I have to say, the difference is a bit surprising, pixi seems to be doing rather poor job with canvas animation.
